python/taichi/tools/video.py

```diff
@@ -32,25 +31,13 @@ def get_ffmpeg_path():
     return 'ffmpeg'
 
 
-def mp4_to_gif(input_fn, output_fn, framerate):
-    # Generate the palette
-    palette_name = 'palette.png'
-    if get_os_name() == 'win':
-        command = get_ffmpeg_path(
-        ) + f" -loglevel panic -i {input_fn} -vf 'palettegen' -y {palette_name}"
-    else:
-        command = get_ffmpeg_path(
-        ) + f" -loglevel panic -i {input_fn} -vf 'fps={framerate}," \
-            f"scale=320:640:flags=lanczos,palettegen' -y {palette_name}"
-    # print command
-    os.system(command)
-
-    # Generate the GIF
-    command = get_ffmpeg_path(
-    ) + f" -loglevel panic -i {input_fn} -i {palette_name} -lavfi paletteuse -y {output_fn}"
-    # print command
-    os.system(command)
-    os.remove(palette_name)
+def mp4_to_gif(input_fn, output_fn, framerate, **kwargs):
+    from moviepy.editor import \
+        VideoFileClip  # pylint: disable=import-outside-toplevel
```

Comment:
Does this lead to a new dependency `moviepy`? That's OK, but you need a `try: import moviepy...` and nofity the user to install, instead of directly throw an import error.

Reply:
Yes it will introduce the dependency on `moviepy`, which further depends on `ffmpeg` and `imageio`.